### ISSUE TITLE
ECL complains that string is too long

### ISSUE THREAD

**Issue by djhaskin987** [fukamachi/quri#76]
When I attempt to load the library and use it in Embeddded Common Lisp on Windows using MSVC compilers, I get this error:
```
;;; C:/Users/bhw/AppData/Local/cache/common-lisp/ecl-21.2.1-0a3c4f53-win-x64/C/Users/bhw/Code/lisp/cl-i/.bundle-libs/s│bhw@DESKTOP-FVT988R MINGW64 ~/Code/lisp/cl-i
oftware/quri-20221106-git/src/etld.data(824): fatal error C1091: compiler limit: string exceeds 65535 bytes in length │$ ls
Condition of type: COMPILE-FILE-ERROR
```
So `etld.data` is being loaded as a string, but it's too big.

Please fix, this is preventing me from using ECL on windows and dexador at the same time :(

**Comment by Ambrevar** [comment]
I can't reproduce on Linux.

The etld.data loading happens in `etld.lisp`, more specifically here:

```lisp
(eval-when (:compile-toplevel :load-toplevel :execute)
  (defvar *default-etld-names*
    #.(asdf:system-relative-pathname :quri #P"data/effective_tld_names.dat"))

  (defun load-etld-data (&optional (etld-names-file *default-etld-names*))
    (with-open-file (in etld-names-file
                        :element-type #+lispworks :default #-lispworks 'character
                        :external-format #+clisp charset:utf-8 #-clisp :utf-8)
      (loop with special-tlds = nil
         with normal-tlds = (make-hash-table :test 'equal)
         with wildcard-tlds = (make-hash-table :test 'equal)
         for line = (read-line in nil nil)
         while line
         unless (or (= 0 (length line))
                    (starts-with-subseq "//" line))
         do (cond
              ((starts-with-subseq "*" line)
               (setf (gethash (subseq line 2) wildcard-tlds) t))
              ((starts-with-subseq "!" line)
               (push (subseq line 1) special-tlds))
              (t
               (setf (gethash line normal-tlds) t)))
         finally (return (list normal-tlds wildcard-tlds special-tlds))))))
```

As you can see, the file is read line by line, so this is not it.

Could you share the rest of your error output?

**Comment by djhaskin987** [comment]
That you can't reproduce this on Linux is expected behavior.

You see, this 64k limitation is a limitation of the Microsoft Visual Studio C++ compiler tool chain. It is a Windows limitation. Further, this limitation is not present even if you use minGW. I might have used MingGW if I knew how to get it to work, but I really would like to use msvc as it is the best way to get a Windows executable.

I had planned on splitting up this file three ways, because the code does that anyway. It splits up the loaded links into three different categories. If we did that the file would be smaller than 64k and I could move on with my life.

What do you guys think of that? I can try to put in a pull request in the next few days.

**Comment by djhaskin987** [comment]
That said I will share the rest of my error output as soon as I get it :)

**Comment by djhaskin987** [comment]
I wrote a fix in #77 , if you could accept that PR, that would be great :) then this would be fixed.

**Comment by Ambrevar** [comment]
Fixed with #77.
